objects/src/block/header.rs

```diff
@@ -275,6 +275,39 @@ impl Deserializable for BlockHeader {
     }
 }
 
+
+
+/// BLOCK NUMBER STRUCT
+
+/// Holds `u32` type  to signify Block Number
+
+
+#[derive(Debug, Eq, PartialEq, Copy, Clone, PartialOrd, Ord)]
+pub struct BlockNumber(u32);
+
+impl Serializable for BlockNumber {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        target.write_u32(self.0);
+    }
+
+    fn get_size_hint(&self) -> usize {
+        core::mem::size_of::<u32>()
+    }
+}
+
+impl From<u32> for BlockNumber {
+    fn from(value: u32) -> Self {
+        BlockNumber(value)
+    }
+}
+
+impl BlockNumber{
+    pub const fn block_epoch_from_number(&self) -> u16 {
```

Comment:
```suggestion
    pub const fn epoch(&self) -> u16 {
```

We can simplify this to just `epoch` because this method lives on `BlockNumber` now and is no longer standalone. `block_number.epoch()` reads nicer.